testUtils/userHelper.ts

```diff
@@ -0,0 +1,160 @@
+import { faker } from '@faker-js/faker';
+import jwt from 'jsonwebtoken';
+
+import { HackneyGoogleUser } from '../domain/HackneyGoogleUser';
+import { HackneyResident } from '../domain/HackneyResident';
+import { HackneyGoogleUserWithPermissions } from '../lib/utils/googleAuth';
+import { generateJWTTokenTestData } from './jwtTokenHelper';
+
+export const AUTHORISED_OFFICER_GROUP_TEST = 'authorized-officer-group';
+export const AUTHORISED_MANAGER_GROUP_TEST = 'authorized-manager-group';
+export const AUTHORISED_ADMIN_GROUP_TEST = 'authorized-admin-group';
+export const AUTHORISED_READONLY_GROUP_TEST = 'authorized-readonly-group';
+export const JWT_TEST_SECRET = 'secret';
+
+const issuedAtInMilliseconds = new Date().getMilliseconds();
+
+export enum UserRole {
+  Officer,
+  Manager,
+  Admin,
+  ReadOnly,
+}
+
+interface UserClaims {
+  hasAdminPermissions: boolean;
+  hasManagerPermissions: boolean;
+  hasOfficerPermissions: boolean;
+  hasReadOnlyPermissions: boolean;
+}
+
+export const getGroupByRole = (role: UserRole): string => {
+  switch (role) {
+    case UserRole.ReadOnly: {
+      return AUTHORISED_READONLY_GROUP_TEST;
+    }
+    case UserRole.Officer: {
+      return AUTHORISED_OFFICER_GROUP_TEST;
+    }
+    case UserRole.Manager: {
+      return AUTHORISED_MANAGER_GROUP_TEST;
+    }
+    case UserRole.Admin: {
+      return AUTHORISED_ADMIN_GROUP_TEST;
+    }
+    default:
+      return '';
+  }
+};
+
+export const getClaimsByRole = (role?: UserRole): UserClaims => {
+  switch (role) {
+    case UserRole.ReadOnly: {
+      return {
+        hasAdminPermissions: false,
+        hasManagerPermissions: false,
+        hasOfficerPermissions: false,
+        hasReadOnlyPermissions: true,
+      };
+    }
+    case UserRole.Officer: {
+      return {
+        hasAdminPermissions: false,
+        hasManagerPermissions: false,
+        hasOfficerPermissions: true,
+        hasReadOnlyPermissions: false,
+      };
+    }
+    case UserRole.Manager: {
+      return {
+        hasAdminPermissions: false,
+        hasManagerPermissions: true,
+        hasOfficerPermissions: false,
+        hasReadOnlyPermissions: false,
+      };
+    }
+    case UserRole.Admin: {
+      return {
+        hasAdminPermissions: true,
+        hasManagerPermissions: false,
+        hasOfficerPermissions: false,
+        hasReadOnlyPermissions: false,
+      };
+    }
+    default:
+      return {
+        hasAdminPermissions: false,
+        hasManagerPermissions: false,
+        hasOfficerPermissions: false,
+        hasReadOnlyPermissions: false,
+      };
+  }
+};
```

Comment:
Could probably reduce its size a lot:

```ts
export const getClaimsByRole = (role?: UserRole): UserClaims => {
  const userPerms = {
        hasAdminPermissions: false,
        hasManagerPermissions: false,
        hasOfficerPermissions: false,
        hasReadOnlyPermissions: false,
      };

   switch (role) {
    case UserRole.ReadOnly: {
      userPerms.hasReadOnlyPermissions = true;
      break;
    }
    case UserRole.Officer: {
      userPerms.hasOfficerPermissions: true;
      break;
    }
    case UserRole.Manager: {
      userPerms.hasManagerPermissions = true;
      break;
    }
    case UserRole.Admin: {
      userPerms.hasAdminPermissions = true;
      break;
    }
    default:
      break; 
  }
  return userPerms;
};
```

Reply:
A lot tidier that way 👍 Updated in  [#0213be2](https://github.com/LBHackney-IT/lbh-housing-register/pull/395/commits/0213be25f6dcb0d99b84e50bf04aa55b15ca655a)